Core/include/Acts/Propagator/RiddersPropagator.ipp

```diff
@@ -197,42 +209,44 @@ Acts::RiddersPropagator<propagator_t>::wiggleDimension(
       }
     }
   }
+
   return derivatives;
 }
 
 template <typename propagator_t>
-auto Acts::RiddersPropagator<propagator_t>::calculateCovariance(
+auto Acts::RiddersPropagator<propagator_t>::calculateJacobian(
+    const std::vector<double>& deviations,
     const std::array<std::vector<Acts::BoundVector>, Acts::eBoundSize>&
-        derivatives,
-    const Acts::BoundSymMatrix& startCov,
-    const std::vector<double>& deviations) const -> Covariance {
+        derivatives) -> Jacobian {
   Jacobian jacobian;
-  jacobian.setIdentity();
-  jacobian.col(eBoundLoc0) = fitLinear(derivatives[eBoundLoc0], deviations);
-  jacobian.col(eBoundLoc1) = fitLinear(derivatives[eBoundLoc1], deviations);
-  jacobian.col(eBoundPhi) = fitLinear(derivatives[eBoundPhi], deviations);
-  jacobian.col(eBoundTheta) = fitLinear(derivatives[eBoundTheta], deviations);
-  jacobian.col(eBoundQOverP) = fitLinear(derivatives[eBoundQOverP], deviations);
-  jacobian.col(eBoundTime) = fitLinear(derivatives[eBoundTime], deviations);
-  return jacobian * startCov * jacobian.transpose();
+  jacobian.col(eBoundLoc0) = fitLinear(deviations, derivatives[eBoundLoc0]);
+  jacobian.col(eBoundLoc1) = fitLinear(deviations, derivatives[eBoundLoc1]);
+  jacobian.col(eBoundPhi) = fitLinear(deviations, derivatives[eBoundPhi]);
+  jacobian.col(eBoundTheta) = fitLinear(deviations, derivatives[eBoundTheta]);
+  jacobian.col(eBoundQOverP) = fitLinear(deviations, derivatives[eBoundQOverP]);
+  jacobian.col(eBoundTime) = fitLinear(deviations, derivatives[eBoundTime]);
+  return jacobian;
 }
 
 template <typename propagator_t>
 Acts::BoundVector Acts::RiddersPropagator<propagator_t>::fitLinear(
-    const std::vector<Acts::BoundVector>& values,
-    const std::vector<double>& deviations) const {
-  BoundVector A;
-  BoundVector C;
-  A.setZero();
-  C.setZero();
+    const std::vector<double>& deviations,
```

Comment:
Do you know where the formulas for this function come from?

Reply:
which one? the linear fit? I think this is a classical linear regression https://en.wikipedia.org/wiki/Linear_regression#Least-squares_estimation_and_related_techniques (without having it checked)

not sure how effective this is for estimating the derivative because I would not expect a slope naively ... but okay